pkg/subctl/resource/schedule_pod.go

```diff
@@ -0,0 +1,186 @@
+/*
+© 2021 Red Hat, Inc. and others.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package resource
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/submariner-io/shipyard/test/e2e/framework"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type NetworkPodType int
+
+type NetworkPodScheduling int
+
+const (
+	GatewayLabel = "submariner.io/gateway"
+)
+
+const (
+	InvalidScheduling NetworkPodScheduling = iota
+	GatewayNode
+	NonGatewayNode
+)
+
+type NetworkingType bool
+
+const (
+	HostNetworking NetworkingType = true
+	PodNetworking  NetworkingType = false
+)
```

Comment:
We can reuse these constants from shipyard framework.

Reply:
I thought about it, but my idea was to have validate_kubeproxy.go (package cmd) to depend ONLY on resource package and not depend on shipyard/framework.

But I can change it if you prefer to have it that way.

Reply:
But you are using `framework.AwaitResultOrError` anyway 😄

Reply:
haha.. no problem. I'll update validate_kubeproxy.go to use framework.XXXX directly